script/check-infra-if-index-changed

```diff
@@ -0,0 +1,148 @@
+    sudo ip link del veth-a || true
+
+    exit $EXIT_CODE
+}
+
+wait_for_socat()
+{
+    if [[ "$(head -n2 "$SOCAT_OUTPUT" | wc -l | tr -d ' ')" == 2 ]]; then
+        RADIO_PTY=$(head -n1 "$SOCAT_OUTPUT" | grep -o '/dev/.\+')
+        return 0
+    else
+        echo 'Still waiting for socat'
+    fi
+    return 1
+}
+
+timeout_run()
+{
+    local count="$1"
+    local exit_code
+    shift 1
+
+    while [[ $count != 0 && $exit_code != 0 ]]; do
+        count=$((count - 1))
+        "$@" && return 0 || exit_code=$?
+        sleep 1
+    done
+
+    return $exit_code
+}
+
+do_build()
+{
+    ./script/cmake-build simulation
+    ./script/cmake-build posix -DOT_PLATFORM_NETIF=1 -DOT_BORDER_ROUTING=ON -DOT_DAEMON=on -DOT_POSIX_EXIT_ON_INFRA_NETIF_LOST=OFF
+}
+
+do_check()
+{
+    trap at_exit INT TERM EXIT
+
+    sudo rm -rf tmp
+
+    # Setup socat for RCP simulation
+    SOCAT_OUTPUT=/tmp/ot-socat
+    socat -d -d pty,raw,echo=0 pty,raw,echo=0 >/dev/null 2>$SOCAT_OUTPUT &
+    timeout_run 10 wait_for_socat
+    echo 'RADIO_PTY' "$RADIO_PTY"
+
+    RADIO_NCP_PATH="$PWD/build/simulation/examples/apps/ncp/ot-rcp"
+    $RADIO_NCP_PATH 1 >"$RADIO_PTY" <"$RADIO_PTY" &
+
+    RADIO_URL="spinel+hdlc+uart://${RADIO_PTY}"
+
+    # Setup virtual ethernet pair
+    sudo ip link add veth-a type veth peer name veth-b
+    sudo ip link set veth-a up
+    sudo ip link set veth-b up
+
+    # Start ot-daemon with veth-a as infra interface
+    OT_DAEMON_LOG=/tmp/ot-daemon.log
+    sudo -E "$PWD/build/posix/src/posix/ot-daemon" -d7 -v "${RADIO_URL}" veth-a 2>&1 | tee "${OT_DAEMON_LOG}" &
+    sleep 3
+
+    OT_CTL_PATH="$PWD/build/posix/src/posix/ot-ctl"
+
+    # Bring up Thread network
+    sudo "${OT_CTL_PATH}" panid 0xface || die 'failed to set panid'
+    sudo "${OT_CTL_PATH}" ifconfig up || die 'failed to bring ifconfig up'
+    sudo "${OT_CTL_PATH}" thread start || die 'failed to start thread'
+    sleep 5
+    sudo "${OT_CTL_PATH}" state | grep -q 'leader' || die 'did not become leader'
+
+    # Enable border routing
+    sudo "${OT_CTL_PATH}" br enable || die 'failed to enable border routing'
+    sleep 2
+    sudo "${OT_CTL_PATH}" br state | grep -q 'enabled' || die 'border routing not enabled'
+
+    # Simulate infra interface index change
+    echo "Simulating infra interface index change..."
+    sudo ip link del veth-a
+    sleep 1
+    sudo ip link add veth-a type veth peer name veth-b
+    sudo ip link set veth-a up
+    sudo ip link set veth-b up
+    sleep 5
+
+    # Verify border routing is still functional by checking for a published prefix
+    # The border router should re-detect the interface and re-establish routing.
+    # A key indicator of this is the successful publication of the on-link prefix.
+    sudo "${OT_CTL_PATH}" br omrprefix | grep -q . || die 'OMR prefix not advertised after infra if index change'
+    sudo "${OT_CTL_PATH}" br onlinkprefix | grep -q . || die 'On-link prefix not advertised after infra if index change'
```

Comment:
I think we should check the address on the host interface.
something like this:
```suggestion
    onlinkprefix="$(sudo "${OT_CTL_PATH}" br onlinkprefix | grep fd)"
    sudo ip addr dev veth-a | grep "${onlinkprefix}" || die 'On-link prefix not advertised after infra if index change'
```